core/core.go

```diff
@@ -436,6 +462,14 @@ func (app *App) Start(ctx context.Context) error {
 		blockWatcherErrChan <- app.blockWatcher.Watch(innerCtx)
 	}()
 
+	if blocksElapsed.Int64() >= maxBlocksStoredInNonArchiveNode {
```

Comment:
This if statement `blocksElapsed.Int64() >= maxBlocksStoredInNonArchiveNode` effectively appears twice. Can we move this chunk of code to immediately after `app.db.ClearAllMiniHeaders()` so we don't have to check if we've exceeded `maxBlocksStoredInNonArchiveNode` twice?

Reply:
I initially had it above, but would rather we re-validate all orders _after_ the BlockWatcher has been started since it _is_ a blocking operation. While we're re-validating orders, further blocks events could be missed that impact the validity of orders.